src/docs/arch-metadata.dox

```diff
@@ -1,19 +1,78 @@
 with it.
 
-A caller of WiredTiger can use WT_SESSION::open_cursor with a \c uri equal to
-\c "meta:" to read the metadata.  Using this interface, metadata can only
-be queried, not changed.  Changes to the metadata are affected by API calls
-such as WT_SESSION::create, WT_SESSION::drop, WT_SESSION::rename.
+@section metadata_checkpoint Storing Checkpoint Metadata
+
+Metadata plays an important role in the WiredTiger checkpointing process. Metadata is used to store
+the latest checkpoint information for each btree in the system. The checkpoint metadata providing a known
+point in time in which WiredTiger can recover from (e.g after a crash or unexpected shutdown).
+The checkpoint metadata for a given btree is stored within the configuration string under its
+associated \c uri key within the \c WiredTiger.wt table. Metadata configuration entries relevant to capturing
+a given btrees checkpoint state include:
+- \c checkpoint: The files checkpoint entries, including their timestamp, transaction ID and write generation.
+- \c checkpoint_lsn: The LSN of the last checkpoint.
+- \c checkpoint_backup_info:  Incremental checkpoint backup information.
+
+Additionally, the metadata table \c WiredTiger.wt is a btree itself that needs to be checkpointed. Since a
+new metadata entry needs to be created for each btree at the time of checkpointing, the metadata table is the last
+file to be checkpointed in the process. Checkpointing the metadata table last is important
+for capturing a consistent metadata state of the btrees in the database at the time of the last checkpoint.
+
+@section metadata_turtle Turtle file
+
+WiredTiger maintains a special turtle file called \c "WiredTiger.turtle" inside the db directory. This containing the
+most recent checkpointing information of the \c WiredTiger.wt metadata file. Using the turtle file, WiredTiger can
+read the latest checkpoint information when recovering the \c WiredTiger.wt table. Recovering the checkpoint
+of the \c WiredTiger.wt table in turn helps restore a consistent view of the checkpointing information for the other btrees in
+the system.
+
+@section metadata_reading Reading Metadata
+
+The contents of the metadata table is ASCII printable. The URIs and configuration values
+are all printable \c C strings.
+
+@subsection metadata_cursor "metadata:" cursor
+
+To read the metadata, a caller can open a cursor with a \c uri equal
+to \c "metadata:". When opening a metadata cursor (through \c WT_SESSION::open_cursor), a user can
+only read the available metadata. Changes to the metadata cannot be written through the metadata
+cursor interface i.e. the metadata cursor is a read-only interface. Metadata is only changed/affected
+by API calls such as WT_SESSION::create, WT_SESSION::drop, WT_SESSION::rename & WT_SESSION::alter.
+
+@subsection metadata_create_cursor "metadata:create" cursor
+
+Additionally, if you want to only read strings relating to creation configurations of the various
```

Comment:
This is the first (of 3) places where you want to avoid 'you'.

Reply:
Thanks for pointing this out, I'll make sure to avoid using 'you' . I've removed/rewored the 3 uses: 6b9cfb62de5c2d5672ac7b6c711fffc6cb49c20f